src/format/KdbxXmlWriter.cpp

```diff
@@ -129,7 +129,9 @@ void KdbxXmlWriter::writeMetadata()
     if (m_kdbxVersion < KeePass2::FILE_VERSION_4) {
         writeBinaries();
     }
-    writeCustomData();
+    if (m_kdbxVersion >= KeePass2::FILE_VERSION_4) {
```

Comment:
`Metadata::customData()` was supported in versions below `KeePass2::FILE_VERSION_4`.

Reply:
True. I verified it by exporting an XML database with KeePass 2.34.
I think now we got everything the way KeePass does. Again, KDBX4 spec is pretty useless.